[IMP] studio: configure export of customizations (#12046)
odoo/documentation: content/applications/studio/models_modules_apps.rst

================================

When you do any customization with Studio, a new module named :guilabel:`Studio customizations` is
added to your database. You can export this module as a ZIP file using the :guilabel:`Studio Export`
function. The module can then be imported into another Odoo database. This may be useful, for
example, when setting up a new module or for training purposes.

.. note::
   Exporting and importing customizations in this way, rather than using the :doc:`standard Odoo

>    export and import<../essentials/export_import_data>` functions, means data is imported in a

Reviewer: ```suggestion
   export and import <../essentials/export_import_data>` functions, means data is imported in a
```
A space is required; otherwise, the :doc: hyperlink won't work.

   logical way. For example, if the module contains customers and sales orders, the customers are
   created first, since these are required for the sales orders to be created.

.. _studio/export-import/export:

Export customizations
---------------------

To export customizations, click the :icon:`oi-studio` :guilabel:`(Toggle Studio)` button on the main
Odoo dashboard, then :guilabel:`Export`, then either:

- download all Studio customizations by clicking the :guilabel:`Export` button; or
- choose what data to export by clicking :ref:`Configure data and demo data to export
  <studio/export-import/export/configure>`.

.. _studio/export-import/export/configure:

Configure data to export
~~~~~~~~~~~~~~~~~~~~~~~~

To select specific models to export, click :guilabel:`New` on the :guilabel:`Studio Export` screen,
then start typing the name of the relevant model or select it from the list.

.. tip::

> 

Reviewer: ```suggestion
```

Tick the following options as relevant:

- :guilabel:`Demo`: if the exported records should be considered as demo data when imported.
- :guilabel:`Attachments`: if attachments related to exported records should be included in the
  export.
- :guilabel:`Updatable`: if the exported records should be able to be updated during a module update.

If necessary, edit the :guilabel:`Domain` to determine which of the model's records should be
exported. To do so, click the :guilabel:`Edit Domain` button or :icon:`fa-caret-right`
:guilabel:`(Modify filter)` then :guilabel:`Edit Domain`, as appropriate. Proceed to make any
required changes.

After configuring a model for export, click :guilabel:`Studio Export` to return to the main screen.
To download a ZIP file with the customizations for all the listed models, click :guilabel:`Export`.

.. note::
   It is not necessary to select one or more models as all listed models will be included in the
   export. To remove a model from the export, select it and click the :icon:`fa-cog`
   :guilabel:`Actions` button then :icon:`fa-trash-o` :guilabel:`Delete`.

Choose whether to only :guilabel:`Include Data` or to also

> :guilabel:`Include Demo Data`, then click the :guilabel:`Export` button.

Reviewer: You could only include demo data. Or no data at all (but then it wouldn't make much sense to be configuring the data to include)

Reviewer 2: In fact, ticking 'Include Demo Data' also ticks 'Include Data' so there's no way to only 'Include Demo Data'. I'm asking Cédric about a scenario where a user might not include any data (odd as that might seem to us). It's true I didn't consider that option.

Reviewer 2: Text updated to explain the three options (nothing ticked, only Include Data, or Include Data + Include Demo Data)

.. image:: models_modules_apps/studio-export.png

>       :alt: Choosing to export both data and demo data

Reviewer: ```suggestion
   :alt: Choosing to export both data and demo data
```
The `:alt:` markup is misaligned with `.. image:::` (it should still work, but it's our "coding style")

.. _studio/export-import/import:

Import customizations
---------------------

.. warning::
   Before importing, make sure the destination database is on the same Odoo version and contains the
   same apps and modules as the source database. Studio does not add the underlying modules as
   dependencies of the exported module.

To import and install Studio customizations in another Odoo database:

- Connect to the destination database.
- Click the :icon:`oi-studio` :guilabel:`(Toggle Studio)` button on the main Odoo dashboard, then
  :guilabel:`Import`.
- Upload the exported ZIP file. If demo data should be imported, tick :guilabel:`Load demo data`.
- Click :guilabel:`Install`.

> 

Reviewer: ```suggestion
```
Only one empty line is necessary :)

================================

When you do any customization with Studio, a new module named :guilabel:`Studio customizations` is
added to your database. Exporting customizations from one database and importing them in another can
be useful, for example, when setting up a new module or for training purposes.

.. note::

   Exporting customizations from Studio is different from the standard Odoo export function, which
   allows you to export data as an Excel or csv file. With :guilabel:`Studio Export`, data is
   exported as a ZIP file, which corresponds to a module.

   When the module is imported in another database, the data is imported in a logical way, e.g. if
   the module contains customers and sales orders, the customers are created first, since these are
   required for the sales orders.

.. _studio/export-import/export:

Export customizations
---------------------

> To export customizations, from the main Odoo dashboard click the **Toggle Studio** button, then
> :guilabel:`Export`.
> 
> You can download a ZIP file containing all Studio customizations by clicking the :guilabel:`Export`
> button. Alternatively, you can choose which data and/or demo data to export by clicking
> :guilabel:`Configure data and demo data to export`.

Reviewer: ```suggestion
To export customizations, click the :icon:`oi-studio` (:guilabel:`Toggle Studio`) button from the main Odoo dashboard, click
:guilabel:`Export`, and then either:

- download all the data directly by clicking the :guilabel:`Export`
button; or
- configure what should be exported first by clicking :ref:`Configure data and demo data to export <studio/export-import/export/configure>`.
```
- We should use [icons](https://www.odoo.com/documentation/18.0/contributing/documentation/rst_guidelines.html#icons) as much as possible. Some are listed [here](https://www.odoo.com/documentation/18.0/developer/reference/user_interface/icons.html), but not all (the Font Awesome ones).
- I think it makes sense to merge the two paragraphs and list the two alternatives as bullet points.
- I added a `:ref:` to the configure data to export heading, which is partly useless given the heading is just below, but I feel it makes it slightly clearer that there are more instructions.
- I removed some repetitions.

.. _studio/export-import/export/configure:

Configure data to export
~~~~~~~~~~~~~~~~~~~~~~~~

> To select specific models to export, on the :guilabel:`Studio Export` screen click
> :guilabel:`New`, then choose the relevant model.

Reviewer: ```suggestion
To select specific models to export, click
:guilabel:`New` on the :guilabel:`Studio Export` screen, then choose the relevant model.
```

.. tip::

>    Click :guilabel:`Preset` to see a list of all models in your database with records that have been

Reviewer 3: @mcsi-odoo To be more accurate, the "Preset" button will select models
that have modified records (from an hard-coded list of preset models to avoid some noise) and the
custom models created with Studio.

Reviewer 2: Will make that update

   modifed using Studio. To configure one of these models for export, click on the model to open it
   and make the required changes.

Tick the following options as relevant:

- :guilabel:`Demo`: if the exported records should be considered as demo data when imported.
- :guilabel:`Attachments`: if attachments related to exported records should be included in the

>   exported.

Reviewer: ```suggestion
  export.
```

> - :guilabel:`Updatable`: if the exported records should be able to updated during a module update.

Reviewer: ```suggestion
- :guilabel:`Updatable`: if the exported records should be able to be updated during a module update.
```

To set or change the rules determining which of the model's records should be exported, click the

> :icon:`fa-caret-right` icon next to :guilabel:`Domain` then click :guilabel:`Edit domain`.

Reviewer: Clicking the carret-right icon is only needed when reopening something that was previously created. Still, it needs to be explained as the UI/behaviour can be confusing.
![image](https://github.com/user-attachments/assets/88a70c4f-08a6-4361-bbd5-20e597528458)

After configuring a model for export, click :guilabel:`Studio Export` to return to the main screen.
To download a ZIP file with the customizations for all the listed models, click :guilabel:`Export`.

.. note::

> 

Reviewer: ```suggestion
```
The empty line isn't necessary (the note will still be displayed as usual with an empty line though)

   It is not necessary to select one or more models as all listed models will be included in the
   export. To remove a model from the export, select it and click

>    :icon:`fa-cog` :guilabel:`Actions` then :icon:`fa-trash-o` :guilabel:`Delete`.

Reviewer: ```suggestion
   the :icon:`fa-cog` (:guilabel:`Actions`) button then :icon:`fa-trash-o` :guilabel:`Delete`.
```

Reviewer 2: Actually, when the user selects one or more models, a new button with [cog + Action] appears. So this is not a case like we looked at where we are adding the label that only appears on hover.

Choose whether to only :guilabel:`Include Data` or to also
:guilabel:`Include Demo Data`, then click the :guilabel:`Export` button.

.. image:: models_modules_apps/studio-export.png
      :alt: Choosing to export both data and demo data

> When the download is complete, click the **X** to close the :guilabel:`Studio Export` window.

Reviewer: ```suggestion
```
Not sure it's necessary to mention (and you can click Cancel or anything else really). It would be best if the dialog box closed itself though :sweat_smile:

.. _studio/export-import/import:

Import customizations
---------------------

.. warning::

>    Before importing, make sure the destination database is the same Odoo version and contains the

Reviewer: ```suggestion
   Before importing, make sure the destination database is on the same Odoo version and contains the
```

   same apps and modules as the source database. Studio does not add the underlying modules as
   dependencies of the exported module.

To import and install Studio customizations in another database, connect to the destination

> database, then, from the main Odoo dashboard click the **Toggle Studio** button then
> :guilabel:`Import`.

Reviewer: ```suggestion
database, click the :icon:`oi-studio` (:guilabel:`Toggle Studio`) button from the main Odoo dashboard, then
:guilabel:`Import`.
```

> Upload the exported ZIP file. If demo data should be imported, tick :guilabel:`Load demo data`.

Reviewer: This could work as a bullet/numbered list:

- Connect to the DB
- Toggle studio and click Import
- Upload the exported ZIP file
- (if demo data should be imported) tick
- click install
